core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java

```diff
@@ -330,4 +303,103 @@ public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmpty
         ? droppedFromStore
         : droppedFromCatalog;
   }
+
+  private boolean importSchema(NameIdentifier identifier) {
+    EntityCombinedSchema combinedSchema = loadCombinedSchema(identifier);
```

Comment:
Why do you need to `loadCombinedSchema` here again?

Reply:
The table may be imported by other thread before we use write lock after we released read lock.